### DISCUSSION TITLE
Multiple hotspots reporting the same IP/port as listen address

### DISCUSSION THREAD

**Discussion by bilalhp** [helium/miner#1671] [Q&A]
Hello,

There are many cases where multiple hotspots are reporting the same IP/port as listen address but the port is actually forwarded to only one of them. For example:

```/ip4/5.39.184.9/tcp/44158|/p2p/11H2VPHjCE75H7rs5Gv6shsMMNDRPg6KmNLVdrvh5GncN2pKdJi|blurry-plastic-chipmunk|
/ip4/5.39.184.9/tcp/44158|/p2p/11QYCnDcV3BMXzBxTg87zbDAZmyFzb4bktp8NWQ6Lbwo3p7qVPE|bent-coconut-bat|
/ip4/5.39.184.9/tcp/44158|/p2p/11RfuLwxKvYqTqV8gWanNnMYvU4gvH89AsaxH1vWKeLjCWErCUS|cheerful-cobalt-bison|
/ip4/5.39.184.9/tcp/44158|/p2p/11ctS2XYVSRJvjYKThvJCeHW13DDGR8aNZAx5jXH4PeSSqdjtRY|dapper-mocha-raven|
/ip4/5.39.184.9/tcp/44158|/p2p/11ddDyEj6CdabudzXbLRTXJ9rJaJ89wHN69DqFrX6PLsCSQrdoK|wild-boysenberry-meerkat|
/ip4/5.39.184.9/tcp/44158|/p2p/11i86JdXidPJoWoPVTAWt4qLipXNEXGiaVymUqqCeCrv9s4L6G7|glorious-rainbow-bat|
/ip4/5.39.184.9/tcp/44158|/p2p/111KXjazhPE4MqE52U5X88oTG7Fwg58CRyUHXUqK9k9HzXVk5ZW|shambolic-shamrock-rooster|
/ip4/5.39.184.9/tcp/44158|/p2p/112QkYKWUafppCGcgznWqfVZvrEWMDuZU4UnzktM5o1ArpDNAsKp|lively-amber-pigeon|
/ip4/5.39.184.9/tcp/44158|/p2p/112uDpWoFMRNC1aCrAhZoTqd5RSJYP1SsMrLaXi2RGBUA71FQYvj|fast-eggshell-sheep|
/ip4/5.39.184.9/tcp/44158|/p2p/1125NasK255ibz6SVkBCTYUo5uPxpEew1Ses5ndJL1aZtEo8YGga|rough-mauve-turtle|
```

When you connect to /ip4/5.39.184.9/tcp/44158, you will see that it is actually blurry-plastic-chipmunk. But when you connect by p2p addresses, for example /p2p/11QYCnDcV3BMXzBxTg87zbDAZmyFzb4bktp8NWQ6Lbwo3p7qVPE you will see in peer book -s the same IP/port.

I think the problem is caused by there is no identify check upon a connection. Connecting to /p2p/11QYCnDcV3BMXzBxTg87zbDAZmyFzb4bktp8NWQ6Lbwo3p7qVPE actually connects to blurry-plastic-chipmunk but the miner thinks it is successfully connected to bent-coconut-bat

I also think the root cause of the problem is somewhere in stungun logic because ideally these miners should create a relay iso thinking they listen on /ip4/5.39.184.9/tcp/44158

This problem is causing 0-witness beacons because if you are challenged by one of these hotspots, it always leads to 0-witness because the witnesses actually connect to the wrong hotspot to report their witnessing. You can see from explorer that the challenges of those miners are always zero witness. Second problem is that they never beacon because they cannot get connected by a challenger.

I am not sure how many devices in the network are impacted but I think this can be easily checked with a query on ETL to see duplicate listen addresses.

**Comment by bilalhp** [comment]
Running the following on ETL yields a sum of 5.2k
```
with tbl as ( select jsonb_array_elements(gwstat.listen_addrs) ->> 0 as ipaddr, count(*) as cnt
from gateway_status gwstat
where gwstat.online = 'online'
group by jsonb_array_elements(gwstat.listen_addrs) ->> 0
order by cnt desc)
select sum (cnt) from tbl where cnt > 1
```
This means around 2.5k devices in the network are not reachable due to this problem.

**Comment by kissste** [comment]
In sys.config:

Could we make listen_addresses (or is it already) configurable/overwrittable ?

{listen_addresses, ["/ip4/0.0.0.0/tcp/44158"]},
